CMake/ITKSetStandardCompilerFlags.cmake

```diff
@@ -241,7 +241,8 @@ function(check_compiler_optimization_flags c_optimization_flags_var cxx_optimiza
     set(InstructionSetOptimizationFlags
        # https://gcc.gnu.org/onlinedocs/gcc-4.8.0/gcc/i386-and-x86_002d64-Options.html
        # NOTE the corei7 release date was 2008
-       -mtune=native # Tune the code for the computer used compile ITK, but allow running on generic cpu archetectures
+       #-mtune=native # Tune the code for the computer used compile ITK, but allow running on generic cpu archetectures
+       -mtune=generic # for reproducible results https://github.com/InsightSoftwareConsortium/ITK/issues/1939
```

Comment:
NOTE: -mtune=generic is fine.  we could also use the equivalent  -mtune=corei7, or equivalently remove this falg all together.

When march=cpu-type  is specified mtune is implied as -mtune=cpu-type as the generic behavior.

The original thought was to "tune" the instruction scheduling for the build host, but restrict the instructions used to 2008 instruction set.

Reply:
> When march=cpu-type is specified mtune is implied as -mtune=cpu-type as the generic behavior.

Yes, agreed, this also my understanding.

However, my understanding from the docs and verifying in the source code:

https://github.com/gcc-mirror/gcc/blob/edd716b6b1caa1a5cb320a8cd7f626f30198e098/gcc/config/i386/i386-c.c#L301

is that `-mtune=corei7` (which could happen by setting `-march=corei7`, or by setting `-mtune=native` when the build is on a `corei7` processor, or by setting `-mtune=corei7`), will tune for `corei7`, and that tuning will be different than `-mtune=generic`.